isis/tests/DistanceTests.cpp

```diff
@@ -0,0 +1,295 @@
+	EXPECT_EQ(dist.kilometers(), Null);
+	EXPECT_EQ(dist.pixels(), Null);
+	EXPECT_EQ(dist.solarRadii(), Null);
+}
+
+
+TEST(DistanceTests, MetersConstructor) {
+	Distance dist(1500500, Distance::Meters);
+	EXPECT_EQ(dist.meters(), 1500500);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+	EXPECT_EQ(dist.pixels(1), 1500500);
+}
+
+
+TEST(DistanceTests, KilometersConstructor) {
+	Distance dist(1500.5, Distance::Kilometers);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_EQ(dist.meters(), 1500500);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+	EXPECT_EQ(dist.pixels(1), 1500500);
+}
+
+
+TEST(DistanceTests, SolarRadiiConstructor) {
+	Distance dist(0.002155922, Distance::SolarRadii);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+	EXPECT_FLOAT_EQ(dist.meters(), 1.5005e+06);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_FLOAT_EQ(dist.pixels(1), 1.5005e+06);
+}
+
+
+TEST(DistanceTests, PixelsConstructor) {
+	Distance dist(1500500, Distance::Pixels);
+	EXPECT_EQ(dist.pixels(1), 1500500);
+	EXPECT_EQ(dist.meters(), 1500500);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 1500.5);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.002155922);
+}
+
+
+TEST(DistanceTests, PixelsPerMeterConstructor) {
+	Distance dist(1500500, 2);
+	EXPECT_EQ(dist.pixels(2), 1500500);
+	EXPECT_EQ(dist.meters(), 750250);
+	EXPECT_FLOAT_EQ(dist.kilometers(), 750.25);
+	EXPECT_FLOAT_EQ(dist.solarRadii(), 0.0010779609);
+}
+
+
+TEST(DistanceTests, CopyConstructor) {
+	Distance origDist(1500.5, Distance::Meters);
+	Distance copiedDist(origDist);
+	ASSERT_FLOAT_EQ(copiedDist.meters(), 1500.5);
+}
+
+
+TEST(DistanceTests, SetMeters) {
+	Distance dist;
+	dist.setMeters(1500500);
+	ASSERT_EQ(dist.meters(), 1500500);
+}
+
+
+TEST(DistanceTests, SetKilometers) {
+	Distance dist;
+	dist.setKilometers(1500500);
+	ASSERT_EQ(dist.kilometers(), 1500500);
+}
+
+
+TEST(DistanceTests, SetSolarRadii) {
+	Distance dist;
+	dist.setSolarRadii(2);
+	ASSERT_EQ(dist.solarRadii(), 2);
+}
+
+
+TEST(DistanceTests, SetPixels) {
+	Distance dist;
+	dist.setPixels(1500500, 2);
+	ASSERT_EQ(dist.pixels(2), 1500500);
+}
+
+
+TEST(DistanceTests, SetNegativeDistance) {
+	try {
+		Distance dist;
+		dist.setMeters(-1);
+		FAIL() << "Expected error message: Negative distances are not supported";
+	}
+	catch(IException &e) {
+	    EXPECT_TRUE(e.toString().contains("Negative distances are not supported")) 
+	    									<< e.toString().toStdString();
+	}
+	catch(...) {
+	   	FAIL() << "Expected error message: Negative distances are not supported";
+	}
+}
+
+
+// TEST(DistanceTests, ToString) {
+// 	Distance dist(1500500, Distance::Meters);
+// 	ASSERT_EQ(dist.toString(), "1500500 meters");
+// }
+
+
+TEST(DistanceTests, IsValidTrue) {
+	Distance dist(1500500, Distance::Meters);
+	ASSERT_TRUE(dist.isValid());
+}
+
+
+TEST(DistanceTests, IsValidFalse) {
+	Distance dist;
+	ASSERT_FALSE(dist.isValid());
+}
+
+
+//should we do all units?
```

Comment:
Since we have checked above that we can convert other units to miles, and each "mathy" test converts all units into miles before calculating, is it safe to say that we don't need to test everyone combination of units in the tests below? I just used meters for everything.

Reply:
This seems okay to me!

Reply:
:+1: